### DISCUSSION TITLE
Why not create bucket in the init steps "The specified bucket does not exist."

### DISCUSSION THREAD

**Discussion by miladjahandideh** [grafana/mimir#2080] [Development]
I get this error and my `querier` pod does not start.

Why doesn't Mimir create the `bucket` in the init steps?

```log
level=warn ts=2022-06-11T06:37:17.880887987Z caller=sanity_check.go:116 msg="Unable to successfully connect to configured object storage (will retry)" err="blocks storage: unable to successfully send a request to object storage: The specified bucket does not exist."
```

and also `ingester` pod

```log
level=warn ts=2022-06-11T06:42:00.00405871Z caller=sanity_check.go:116 msg="Unable to successfully connect to configured object storage (will retry)" err="blocks storage: unable to successfully send a request to object storage: The specified bucket does not exist."
```

**Comment by pracucci** [comment]
I don't think it should be Mimir responsability to create the bucket. For example, there are cases where Mimir would detect a bucket as not existing, but we still don't want to create it. For example, think about the bucket exist but it's misconfigured in Mimir: we don't want Mimir to create it.

Also, Mimir should run with restricted privileges and, in my opinion, shouldn't expect to be granted privileges to create buckets.

**Reply by miladjahandideh** [reply]
Thank you I goy the point. But I think it cloud be clear in the documentation maybe if it is not and also an AWS S3 example values for helm can be very useful. I will try to add one other S3 backend vaules.yml example other than Minio for the helm.

**Reply by pracucci** [reply]
Agree! Opened an issue here:
https://github.com/grafana/mimir/issues/2113

**Reply by asyncjason** [reply]
I started with the "Play with Mimir" docker-compose file and I'm pretty sure mimir creates the bucket for blocks_storage doesn't it? I was confused as to why it didn't for ruler too.
If the functionality already exists, then why not make it a configuration option for ruler too? For those that are using minio to get ruler API functionality.
Is my only option to build a side car just to get ruler working with minio?

**Reply by asyncjason** [reply]
Here's a way to create one with a sidecar for minio:
https://stackoverflow.com/questions/66412289/minio-add-a-public-bucket-with-docker-compose
